pennylane/transforms/intermediate_reps/rowcol.py

```diff
@@ -281,27 +290,31 @@ def _eliminate(P: np.ndarray, connectivity: nx.Graph, idx: int, mode: str, verbo
         # ... and no constraints for second pass (i.2.3)
         _ = [state := _update(*state, child, parent) for child, parent in visit_nodes]
 
-    if verbose:
-        print(f"CNOTs from {mode} elimination ({idx}): {cnots}")
     return P % 2, cnots
 
 
-def rowcol(P: np.ndarray, connectivity: nx.Graph = None, verbose: bool = False) -> list[tuple[int]]:
-    r"""CNOT routing algorithm ROWCOL.
+@transform
+def rowcol(
+    tape: QuantumScript, connectivity: nx.Graph = None
+) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    r"""CNOT routing algorithm `ROWCOL <https://pennylane.ai/compilation/rowcol-algorithm>`__.
 
     This algorithm was introduced by `Wu et al. <https://arxiv.org/abs/1910.14478>`__ and is
     detailed in the `compilation hub <https://pennylane.ai/compilation/rowcol-algorithm>`__,
-    where examples can be found as well.
+    where additional examples can be found as well.
 
     Args:
-        P (np.ndarray): Parity matrix to implement. Will not be altered
+        tape (QNode or QuantumScript or Callable): Input circuit containing only :class:`~.CNOT` gates. Will internally be translated to the :func:`~.parity_matrix` IR.
         connectivity (nx.Graph): Connectivity graph to route into. If ``None`` (the default),
```

Comment:
Can the networkx connectivity graph accept any wire labels? Like wire `'a'`? Or is it limited to integers?